src/storage/db_interface_frontend.py

```diff
@@ -277,6 +277,28 @@ def find_missing_files(self):
                 parent_to_included.pop(parent_uid)
         return parent_to_included
 
+    def find_orphaned_objects(self):
+        ''' find File Objects whose parent firmware is missing '''
+        orphans_by_parent = {}
+        query_result = list(self.file_objects.aggregate([
+            {'$unwind': '$parent_firmware_uids'},
+            {'$group': {'_id': 0, 'all_parent_uids': {'$addToSet': '$parent_firmware_uids'}}}
+        ], allowDiskUse=True))
+        if query_result:
+            fo_parent_firmware = set(query_result[0]['all_parent_uids'])
+            missing_uids = fo_parent_firmware.difference(self._get_all_firmware_uids())
+            if missing_uids:
+                query_result = self.file_objects.find({'parent_firmware_uids': {'$in': list(missing_uids)}})
+                for fo_entry in query_result:
+                    for uid in missing_uids:
+                        if uid in fo_entry['parent_firmware_uids']:
+                            orphans_by_parent.setdefault(uid, []).append(fo_entry['_id'])
+        return orphans_by_parent
+
+    def _get_all_firmware_uids(self) -> List[str]:
+        query_result = list(self.firmwares.aggregate([{'$group': {'_id': 0, 'firmware_uids': {'$push': '$_id'}}}]))
```

Comment:
Why use aggregate here? Does it bring a speed up over sth. like 
```python3
    def _get_all_firmware_uids(self) -> List[str]:
        return [
            firmware['_id'] for firmware in self.firmwares.find({}, {' _id': 1})
        ]
```

Reply:
I was unsure about the speed and efficiency, so I tried both with these results:

> find: 0.001560211181640625 seconds
> aggregate: 0.0003306865692138672 seconds

So the aggregate seems actually to be faster (about 4.7x) (probably because the iteration is done in MongoDB and not in Python).

Reply:
Fancy shit. So I have to accept this.